### PR TITLE
Assignment 2 - extract version route to plugin

### PR DESCRIPTION
**Pull request by toboid**
#43

### CODE REVIEW

File: lib/version.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+// Declare internals
+
+var internals = {};
+
+
+internals.versionInfo = { version: Package.version };
+
+module.exports.register = function (server, options, next) {
```
Comment (chyld) [inline lib/version.js]: you can just say exports.register, no need to add module

File: lib/version.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                return reply(internals.versionInfo);
+            }
+        }
+    });
+
+    return next();
+};
+
+
+module.exports.register.attributes = {
```
Comment (chyld) [inline lib/version.js]: remove module, just exports.register

File: lib/index.js [hunk 3]
```diff
@@ -15,22 +15,14 @@ internals.init = function () {
     var server = new Hapi.Server();
     server.connection({ port: 8000 });
 
-    server.route({
-        method: 'GET',
-        path: '/version',
-        config: {
-            description: 'Returns the version of the server',
-            handler: function (request, reply) {
+    server.register(Version, function (registerErr) {
```
Comment (hueniverse) [inline lib/index.js]: Unless you need to access errors on multiple levels, just call it `err`. It's a standard.

File: lib/version.js (hunk 1 above)
Comment (toboid) [inline lib/version.js]: thanks

File: lib/index.js (hunk 3 above)
Comment (toboid) [inline lib/index.js]: will do, thanks

**Pull request closed without merging**